View controller removal bug

edrutter-je opened the pull request:
Fix bug where a ViewController's view was not removed from it's superview when calling remove(viewController: UIViewController) if it was embedded in a 'padding view'.

edrutter-je committed:
Fix issue #20 where view was not removed when removing view controller

edrutter-je committed:
Increment patch version [4.0.1]

stevenwarren-je commented:
If we only call removeFromSuperview on the view controllers view, the 'padding view' will remain in the view hierarchy.

arrangedView.removeFromSuperview() will take care of it all, including removing arrangedView from arrangedSubviews.

stevenwarren-je committed:
padding view now also removed from superview if present.

removeFromSuperview also removes the view from the arrangedSubviews array, so that call is no longer needed.

stevenwarren-je committed:
fixed unit test.

stevenwarren-je committed:
Handle the case where there is still a strong reference to the view controller

stevenwarren-je committed:
Updated Unit Test to demonstrate both VC superviews are nil.

stevenwarren-je commented:
@edrutter-je would appreciate it if you gave this a wee once over :)

edrutter-je commented:
LGTM
